app/services/base_service.rb

```diff
@@ -4,13 +4,32 @@ class BaseService
   include AfterCommitEverywhere
 
   class FailedResult < StandardError
-    attr_reader :result
+    attr_reader :result, :original_error
 
-    def initialize(result, message)
+    def initialize(result, message, original_error: nil)
       @result = result
+      @original_error = original_error
 
       super(message)
     end
+
+    def original_error_details
+      if original_error.is_a?(::Stripe::StripeError)
+        stripe_error_details(original_error)
+      end
+    end
+
+    private
+
+    def stripe_error_details(err)
```

Comment:
In my opinion this should not be handled in the generic failed result, but passed when creating the result itself (maybe via a decorator?).
I'm pretty sure that otherwise we will will soon have a LOT of error specific handling in this class

Reply:
> we will will soon have a LOT of error specific handling in this class

Yes probably, but I don't see this as an issue ^^

Reply:
Refactoring! 📈